playgrounds/react-vite/src/tabs/Shop.tsx

```diff
@@ -0,0 +1,128 @@
+import { Text } from '@0xsequence/design-system';
+import {
+	ShopCollectibleCard,
+	useCollection,
+	useListTokenMetadata,
+} from '@0xsequence/marketplace-sdk/react';
+import type { Address } from 'viem';
+import { useReadContract } from 'wagmi';
+import {
+	type ContractType,
+	ERC1155_SALES_CONTRACT_ABI,
+} from '../../../../sdk/src';
+import { useList1155SaleSupplies } from '../../../../sdk/src/react/hooks/useList1155SaleSupplies';
+
+// Helper function to extract cost from token sale details with type safety
+const getSaleCost = (
+	result:
+		| `0x${string}`
+		| {
+				cost: bigint;
+				supplyCap: bigint;
+				startTime: bigint;
+				endTime: bigint;
+				merkleRoot: `0x${string}`;
+		  }
+		| undefined,
+) => {
+	if (!result || typeof result !== 'object' || !('cost' in result)) {
+		return undefined;
+	}
+	return result.cost.toString();
+};
+
+export function Shop() {
+	const tokenIds = ['1', '2', '3', '10'];
+	const chainId = 80002;
+	const contractAddress: Address = '0x98d2dd98e762492435c731346c799145d4e61e5b';
+	const salesContractAddress: Address =
+		'0xddc7029ce8390cdd6b6c1ff58d4bf4c3f1f88bed';
+
+	const { data: collection, isLoading: collectionIsLoading } = useCollection({
+		chainId,
+		collectionAddress: contractAddress,
+	});
+
+	const { data: tokenMetadataList, isLoading: tokenMetadataListLoading } =
+		useListTokenMetadata({
+			chainId,
+			contractAddress,
+			tokenIds,
+		});
+
+	const { extendedSupplyData, getSupply, supplyDataLoading } =
+		useList1155SaleSupplies({
+			tokenIds,
+			salesContractAddress,
+		});
+
+	const { data: paymentCurrencyAddress, isLoading: paymentCurrencyIsLoading } =
```

Comment:
`paymentCurrencyAddress` is `undefined` and `extendedSupplyData` is `[]`  for some reason... 🤔

Reply:
I did not think about it... But you might be on the wrong chain
```suggestion
	const { data: paymentCurrencyAddress, isLoading: paymentCurrencyIsLoading } =		
	useReadContract({
	    chainId
			address: salesContractAddress,
			abi: ERC1155_SALES_CONTRACT_ABI,
			functionName: 'paymentToken',
		});
```